src/zenml/artifacts/external_artifact_config.py

```diff
@@ -32,26 +31,23 @@ class ExternalArtifactConfiguration(BaseModel):
     """
 
     id: Optional[UUID] = None
-    name: Optional[str] = None
-    version: Optional[str] = None
-    model: Optional[Model] = None
 
-    @model_validator(mode="after")
-    def external_artifact_validator(self) -> "ExternalArtifactConfiguration":
-        """Model validator for the external artifact configuration.
+    @model_validator(mode="before")
+    @classmethod
+    @before_validator_handler
+    def _remove_old_attributes(cls, data: Dict[str, Any]) -> Dict[str, Any]:
+        """Remove old attributes that are not used anymore.
 
-        Raises:
-            ValueError: if both version and model fields are set.
+        Args:
+            data: The model data.
 
         Returns:
-            the validated instance.
+            Model data without the removed attributes.
         """
-        if self.version and self.model:
-            raise ValueError(
-                "Cannot provide both `version` and `model` when "
-                "creating an external artifact."
-            )
-        return self
+        data.pop("name", None)
+        data.pop("version", None)
+        data.pop("model", None)
+        return data
```

Comment:
It looks like older pipelines might fail silently downstream because of this.

Reply:
In which case would they fail?

I'm guessing this would only be possible if upgrading the server to the new release while still having an old scheduled pipeline running that uses a Docker image of an older version? The server would then remove those attributes before sending the response.

Would that case be fixed if I just don't remove this data here and add `extra=allow` in the pydantic config?

Reply:
To me, it would be better just to give some instructions on how to fix this and write that deprecation clearly in the new release notes. I don't ask for any technical solution here - these guys should have been gone for a while already.

Reply:
The problem here is that these attributes will already be removed on the server side, which means the warning/instructions will be very hidden. I instead added some instructions on the runtime error below that happens if the ID is missing.